supplier/views.py

```diff
@@ -0,0 +1,44 @@
+from django.shortcuts import render
+from supplier.models import Supplier
+from .forms import SupplierForm
+
+
+# def index(request):
+#     form = SupplierForm()
+#     context = {'form':form}
+#     return render(request, 'supplier/supplier_sign_up.html',context)
+# def login_page(request):
+#     return render(request, 'supplier/supplier_sign_up.html')
+
+
+def signup_supplier(request):
+    form = SupplierForm()
+    context = {'form': form, 'message': ['username should be longer than 6',
+                                         'password should be longer than 6',
+                                         'business name should be longer than 6']}
+    if request.method == 'POST':
+        form = SupplierForm(request.POST)
+        user_name = request.POST.get('user_name')
+        password = request.POST.get('password')
+        first_name = request.POST.get('first_name')
+        last_name = request.POST.get('last_name')
+        business_name = request.POST.get('business_name')
+        supplier = Supplier(user_name=user_name,
+                            password=password,
+                            first_name=first_name,
+                            last_name=last_name,
+                            business_name=business_name)
+        if form.is_valid():
+            Supplier.save_supplier(supplier)
```

Comment:
what about the option for validation error? shouldnt this line also be in try?

Reply:
The form is valid there is no errors or exception look at the else .